test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs

```diff
@@ -478,6 +478,35 @@ public class InlineConstraintTests
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
         }
 
+        [Theory]
+        // Testing custom inline constraint configured via MapRoute
+        [InlineData("software", HttpStatusCode.OK, "software From Software Controller Index")]
+        // Testing custom inline constraint updated in ConstraintMap
+        [InlineData("hardware", HttpStatusCode.NotFound, null)]
+        [InlineData("HARDWARE", HttpStatusCode.OK, "HARDWARE From Hardware Controller Index")]
+        // Testing custom inline constraint configured with Attribute Routing
+        [InlineData("consumable", HttpStatusCode.OK, "consumable From Consumable Controller Index")]
+        public async Task CustomInlineConstraint_Add_Update(
```

Comment:
Think about if it can be tested via a unit-test. We want to avoid creating functional tests for scenarios that throw unless it's important.